### PR TITLE
new branch for carousel & new pull request

### PR DESCRIPTION
**Pull request by ileenshaar**

### CODE REVIEW

File: src/components/carousel/category-titles.jsx
Comment (mahmoud-alragabi) [inline src/components/carousel/category-titles.jsx:1]: why is the component name different from the file name, I prefer the titles one

File: src/components/carousel/category-titles.jsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import Typography from "@mui/material/Typography";
```
Comment (mahmoud-alragabi) [inline src/components/carousel/category-titles.jsx]: these three imports should be only one

File: src/components/carousel/category-titles.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    >
+      <Box
+        sx={{
+          maxHeight: "70%",
+          overflow: "hidden",
+          padding: "5% 4% 0",
+        }}
+      >
+        <Typography fontSize={60} fontWeight={"900"} lineHeight={"50px"}>
+          {props.title}
```
Comment (mahmoud-alragabi) [inline src/components/carousel/category-titles.jsx]: destructure the title and description from props instead of using dot notation

**Review by mahmoud-alragabi — CHANGES_REQUESTED** [review]

File: src/components/carousel/carousel-items.jsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
+import { Paper } from "@mui/material";
```
Comment (mahmoud-alragabi) [inline src/components/carousel/carousel-items.jsx]: again make the imports from mui/material into one

File: src/components/carousel/carousel-items.jsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  left: "40%",
+  width: "60%",
+  height: "80%",
+  top: "10%",
+  borderRadius: "15px 0 0 15px",
+  backgroundColor: "rgba(255, 255, 255, 0.5)",
+  color: "#1B4B66",
+};
+
+function CarouselItem(props) {
```
Comment (mahmoud-alragabi) [inline src/components/carousel/carousel-items.jsx]: destructor props

**Review by mahmoud-alragabi — CHANGES_REQUESTED** [review]

File: src/components/carousel/carousel-items.jsx
Comment (mahmoud-alragabi) [inline src/components/carousel/carousel-items.jsx:1]: remove the s at the end of the file name

File: src/components/carousel/carousel-items.jsx [hunk 5]
```diff
@@ -22,21 +23,23 @@ function CarouselItem(props) {
...
-            ...imageContainerStyles,
-          }}
-        >
-          <ProductImage src={props.item.imgPath} width="100%" height={300} />
-          <CategoryItems
-            title={props.item.name}
-            description={props.item.description}
-          />
-        </Box>
+        <Link href="#">
```
Comment (mahmoud-alragabi) [inline src/components/carousel/carousel-items.jsx]: why are you wrapping the whole carousel with a link, not the whole image should be clickable but only the "see more" button

File: src/components/carousel/carousel-items.jsx [hunk 6]
```diff
@@ -3,6 +3,7 @@ import Box from "@mui/material/Box";
 import EastIcon from "@mui/icons-material/East";
 import { ProductImage } from "../product-image";
 import { CategoryItems } from "./category-titles";
+import Link from "@mui/material/Link";
```
Comment (mahmoud-alragabi) [inline src/components/carousel/carousel-items.jsx]: is there a reason to use Link from mui and not the router link ?

**Review by mahmoud-alragabi — CHANGES_REQUESTED** [review]

File: src/components/carousel/carousel.jsx [hunk 7]
```diff
@@ -2,48 +2,20 @@ import React from "react";
...
-      imgPath:
-        "https://images.unsplash.com/photo-1537996194471-e657df975ab4?auto=format&fit=crop&w=400&h=250",
-    },
-
-    {
-      name: "Random Name #3",
-      description: "Probably the most random thing you have ever seen!",
-      imgPath:
-        "https://images.unsplash.com/photo-1512341689857-198e7e2f3ca8?auto=format&fit=crop&w=400&h=250&q=60",
+  const carouselStyle = (interval) => ({
```
Comment (mahmoud-alragabi) [inline src/components/carousel/carousel.jsx]: this doesn't need to be a function just pass the interval  as a prop and use it and keep the other props as internal state

**Review by mahmoud-alragabi — CHANGES_REQUESTED** [review]

File: src/components/carousel/carousel-items.jsx (hunk 6 above)
Comment (ileenshaar) [reply src/components/carousel/carousel-items.jsx]: I've read this:
Integration with Router Library: If you're using a router library like React Router, it's generally recommended to use that library's Link component. This ensures proper integration with the router's navigation system and allows for features like client-side routing without full page reloads.
so I will change it to router link

**Review by mahmoud-alragabi — DISMISSED** [review]

**Review by mahmoud-alragabi — APPROVED** [review]

**Review by ferassmohana — APPROVED** [review]
ابلعي أبروفد

**Pull request merged**